easybuild/framework/easyconfig/types.py

```diff
@@ -349,6 +349,31 @@ def to_list_of_strings_and_tuples(spec):
     return str_tup_list
 
 
+def to_list_of_strings_and_tuples_and_dicts(spec):
+    """
+    Convert a 'list of lists and strings' to a 'list of dicts and tuples and strings'
```

Comment:
`Convert a 'list of dicts and tuples/lists and strings' to a 'list of dicts and tuples and strings'`